command/healthcheck/pki_audit_visibility.go

```diff
@@ -0,0 +1,186 @@
+	"province",
+	"street_address",
+	"postal_code",
+	"permitted_dns_domains",
+	"policy_identifiers",
+	"ext_key_usage_oids",
+}
+
+var VisibleRespParams = []string{
+	"certificate",
+	"issuing_ca",
+	"serial_number",
+	"error",
+	"ca_chain",
+}
+
+var HiddenReqParams = []string{
+	"private_key",
+	"pem_bundle",
+}
+
+var HiddenRespParams = []string{
+	"private_key",
+	"pem_bundle",
+}
+
+type AuditVisibility struct {
+	Enabled            bool
+	UnsupportedVersion bool
+
+	IgnoredParameters map[string]bool
+	TuneData          map[string]interface{}
+}
+
+func NewAuditVisibilityCheck() Check {
+	return &AuditVisibility{
+		IgnoredParameters: make(map[string]bool),
+	}
+}
+
+func (h *AuditVisibility) Name() string {
+	return "audit_visibility"
+}
+
+func (h *AuditVisibility) IsEnabled() bool {
+	return h.Enabled
+}
+
+func (h *AuditVisibility) DefaultConfig() map[string]interface{} {
+	return map[string]interface{}{
+		"ignored_parameters": []string{},
+	}
+}
+
+func (h *AuditVisibility) LoadConfig(config map[string]interface{}) error {
+	var err error
+
+	coerced, err := stringList(config["ignored_parameters"])
+	if err != nil {
+		return fmt.Errorf("error parsing %v.ignored_parameters: %v", h.Name(), err)
+	}
+	for _, ignored := range coerced {
+		h.IgnoredParameters[ignored] = true
+	}
+
+	h.Enabled, err = parseutil.ParseBool(config["enabled"])
+	if err != nil {
+		return fmt.Errorf("error parsing %v.enabled: %w", h.Name(), err)
+	}
+
+	return nil
+}
+
+func (h *AuditVisibility) FetchResources(e *Executor) error {
+	exit, _, data, err := fetchMountTune(e, func() {
+		h.UnsupportedVersion = true
+	})
+	if exit {
+		return err
+	}
+
+	h.TuneData = data
+
+	return nil
+}
+
+func (h *AuditVisibility) Evaluate(e *Executor) (results []*Result, err error) {
+	if h.UnsupportedVersion {
+		// Shouldn't happen; /certs has been around forever.
+		ret := Result{
+			Status:   ResultInvalidVersion,
+			Endpoint: "/{{mount}}/certs",
+			Message:  "This health check requires Vault 1.11+ but an earlier version of Vault Server was contacted, preventing this health check from running.",
+		}
+		return []*Result{&ret}, nil
+	}
+
+	sourceMap := map[string][]string{
+		"audit_non_hmac_request_keys":  VisibleReqParams,
+		"audit_non_hmac_response_keys": VisibleRespParams,
+	}
+	for source, visibleList := range sourceMap {
+		actual, err := stringList(h.TuneData[source])
+		if err != nil {
+			return nil, fmt.Errorf("error parsing %v from server: %v", source, err)
+		}
+
+		for _, param := range visibleList {
+			found := false
+			for _, tuned := range actual {
+				if param == tuned {
+					found = true
+					break
+				}
+			}
+
+			if !found {
+				ret := Result{
+					Status:   ResultInformational,
+					Endpoint: "/sys/mounts/{{mount}}/tune",
+					Message:  fmt.Sprintf("Mount currently HMACs %v because it is not in %v; as this is not a sensitive security parameter, it is encouraged to disable HMACing to allow better auditing of the PKI engine.", param, source),
```

Comment:
Adding `certificate` here might be touchy with the possibility of the 500 error on syslog audit events being too large if the CRL is large or did I misunderstand that bug report?

Reply:
OTOH, we need to audit `certificate` in order to provide an audit trail of leaf issuance. Its hard, but in general we should recommend `file` type audit devices to avoid this limitation of syslog.